src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/XmlReaderWrapper.cs

```diff
@@ -13,311 +13,86 @@ internal class XmlReaderWrapper : XmlReader
     {
         private XmlReader _reader;
 
-        private Func<XmlReaderWrapper, Task<string>> _getValueFunc;
-        private Func<XmlReaderWrapper, Task<XmlNodeType>> _moveToContentFunc;
-        private Func<XmlReaderWrapper, Task> _skipFunc;
-        private Func<XmlReaderWrapper, Task<bool>> _readFunc;
-        private Func<XmlReaderWrapper, Task<string>> _readInnerXmlFunc;
-
-        private XmlReaderWrapper(XmlReader reader)
+        public static XmlReader CreateFromReader(XmlReader reader)
         {
-            if (reader == null)
-            {
-                throw new ArgumentNullException(nameof(reader));
-            }
-
-            _reader = reader;
-
-            if (_reader.Settings.Async)
-            {
-                InitAsync();
-            }
-            else
+            if (reader is XmlReaderWrapper || reader.Settings.Async)
```

Comment:
Can `reader.Settings` be null? One of the tests being added failed due to `reader.Settings` was null.